### ISSUE TITLE
List Item number with circular background color

### ISSUE THREAD

**Issue by abhaywawale** [noties/Markwon#351]
* **Markwon version**: _4.6.2_

1. Please specify expected/actual behavior

I have below markdown
```
1. How are you
2. I am fine
3. Okay
```
And it renders like below
![Screenshot_20210406-225034_Constitution of India](https://user-images.githubusercontent.com/36884827/113752853-0f3d5e00-972b-11eb-9533-1dd9eec3b136.jpg)

2. Please specify conditions/steps to reproduce (layout, code, markdown used, etc)

I want something like this...
![Untitled](https://user-images.githubusercontent.com/36884827/113753608-e5d10200-972b-11eb-8df4-846b921e16ea.png)

Basically List Item number with a circular background and without the dot.

Help is much appreciated!!!

**Comment by noties** [comment]
Hello @abhaywawale ,

you can achieve that by creating an own span that draws number and its background. You can look at the [OrderedListItemSpan](https://github.com/noties/Markwon/blob/c4cb4420b1dd37ae24d87d4d801f703cd2872141/markwon-core/src/main/java/io/noties/markwon/core/spans/OrderedListItemSpan.java) in the `core` module and adjust it to also draw background. Then you need to change default span type for ordered lists. Take a look at the [sample](https://github.com/noties/Markwon/blob/205ae7b47a66d7bd33a0b21c7c9105929cd8a117/app-sample/src/main/java/io/noties/markwon/app/samples/ChangeBulletSpanSample.java) for a reference that changes span for bullet lists. Basically you will need something like this:

```kotlin
// keep track of original factory for fallback
val original = builder.getFactory(ListItem::class.java)

builder.setFactory(ListItem::class.java) { configuration, props ->
  // both bullet and ordered lists share the same `ListItem` node type
  if (CoreProps.ListItemType.ORDERED == CoreProps.LIST_ITEM_TYPE.get(props)) {
    val number = CoreProps.ORDERED_LIST_ITEM_NUMBER.require(props)
    MyOrderedListItemSpan(number)
  } else {
    original?.getSpans(configuration, props)
  }
}
```
